File: src/array/array_java/Ch7_RotateAnArray.java
package array.array_java;

import java.util.Scanner;

//wrong code check again
public class Ch7_RotateAnArray {
    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);
        System.out.println("Enter No. n:");
        int n = scn.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter array");
        for (int i = 0; i < arr.length; i++) {
            arr[i] = scn.nextInt();
        }

        System.out.println("Enter No. K:");
        int k = scn.nextInt();

        reverse(arr, 0, k - 1);
        rotate(arr, k);
        display(arr);


    }

    public static void reverse(int[] arr, int i, int j) {
        int li = i;
        int ri = j;

        while (li < ri) {
            int temp = arr[li];
            arr[li] = arr[ri];
            arr[ri] = temp;

            li++;
            ri--;
        }
    }

    public static void rotate(int[] arr, int k) {

        k = k % arr.length;
        if (k < 0) {
            k = k + arr.length;
        }
        reverse(arr, 0, arr.length - k - 1);
        reverse(arr, arr.length - k, arr.length - 1);
        reverse(arr, 0, arr.length - 1);
    }

    public static void display(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}

File: src/array/array_kotlin/Ch7_RotateAnArray.kt
package array.array_kotlin

import java.util.*

//wrong code check again
fun main() {
    rotateAnArray()
}

fun rotateAnArray() {
    val scn = Scanner(System.`in`)
    println("Enter No. n:")
    val n = scn.nextInt()
    val arr = IntArray(n)
    println("Enter array")
    for (i in arr.indices) {
        arr[i] = scn.nextInt()
    }
    println("Enter No. K:")
    val k = scn.nextInt()
    reverse(arr, 0, k)
    rotate(arr, k)
    display(arr)
}

fun reverse(arr: IntArray, i: Int, j: Int) {
    var li = i
    var ri = j
    while (li < ri) {
        val temp = arr[li]
        arr[li] = arr[ri]
        arr[ri] = temp
        li++
        ri--
    }
}

fun rotate(arr: IntArray, k: Int) {
    var k = k
    k %= arr.size
    if (k < 0) {
        k += arr.size
    }
    reverse(arr, 0, arr.size - k - 1)
    reverse(arr, arr.size - k, arr.size - 1)
    reverse(arr, 0, arr.size - 1)
}

fun display(arr: IntArray) {
    for (i in arr.indices) {
        print(arr[i].toString() + " ")
    }
}